src/arch/riscv/insts/vector.hh

```diff
@@ -596,6 +596,125 @@ class VlFFTrimVlMicroOp : public VectorMicroInst
         const override;
 };
 
+class VlSegMacroInst : public VectorMemMacroInst
+{
+  protected:
+    VlSegMacroInst(const char* mnem, ExtMachInst _machInst,
+                   OpClass __opClass, uint32_t _vlen)
+        : VectorMemMacroInst(mnem, _machInst, __opClass, _vlen)
+    {}
+
+    std::string generateDisassembly(
+            Addr pc, const loader::SymbolTable *symtab) const override;
+};
+
+class VlSegMicroInst : public VectorMicroInst
+{
+  protected:
+    Request::Flags memAccessFlags;
+    uint8_t regIdx;
+
+    VlSegMicroInst(const char *mnem, ExtMachInst _machInst,
+                   OpClass __opClass, uint32_t _microVl,
+                   uint32_t _microIdx, uint32_t _numMicroops,
+                   uint32_t _field, uint32_t _numFields,
```

Comment:
Should we need to keep the parameters `_numMicroops`, `_field`, `_numFields` in the `VlSegMicroInst`?

Reply:
Hi @rogerchang23424 ,

In principle, I think we need them:

- `_numMicroops` and `_field` are needed to set the registers where the loads are going to be placed.
- `_numFields` is needed to check if the LMUL / NFIELDS combination is allowed. Maybe we could take this NFIELDS from `_machInst` inside the VlSegMicroExecute, but not sure if this is beneficial.

Let me know if you have in mind a better way of doing this.

Thanks!

Reply:
> * `_numFields` is needed to check if the LMUL / NFIELDS combination is allowed. Maybe we could take this NFIELDS from `_machInst` inside the VlSegMicroExecute, but not sure if this is beneficial.

I would like to keep the `_numFields` in it.

Maybe I didn’t make it clear enough. There are duplicate fileds `_numMicroops`, `_field`, `_numFields` in the `VlSegMicroInst` and its derived class.

Thanks.